File: app/src/main/java/com/simon/harmonichackernews/network/VolleyErrorUtils.java
package com.simon.harmonichackernews.network;

import com.android.volley.VolleyError;

final class VolleyErrorUtils {

    private VolleyErrorUtils() {
    }

    static String describe(VolleyError error) {
        if (error == null) {
            return "unknown VolleyError";
        }
        String status = error.networkResponse == null
                ? "noNetworkResponse"
                : "statusCode=" + error.networkResponse.statusCode;
        return error.getClass().getSimpleName() + ", " + status + ", message=" + error.getMessage();
    }
}

File: app/src/main/java/com/simon/harmonichackernews/network/VolleyErrorUtils.kt
package com.simon.harmonichackernews.network

import com.android.volley.VolleyError

internal object VolleyErrorUtils {
    fun describe(error: VolleyError?): String {
        error ?: return "unknown VolleyError"
        val status = error.networkResponse?.let { "statusCode=${it.statusCode}" }
            ?: "noNetworkResponse"
        return "${error.javaClass.simpleName}, $status, message=${error.message}"
    }
}
